builder/builder/relay_aggregator.go

```diff
@@ -81,6 +81,56 @@ func (r *RemoteRelayAggregator) SubmitBlockWithProofs(msg *common.VersionedSubmi
 	return nil
 }
 
+func (r *RemoteRelayAggregator) GetDelegationsForSlot(nextSlot uint64) (types.SignedDelegations, error) {
+	delegationsCh := make(chan *types.SignedDelegations, len(r.relays))
+
+	for i, relay := range r.relays {
+		go func(relay IRelay, relayI int) {
+			delegations, err := relay.GetDelegationsForSlot(nextSlot)
+			if err != nil {
+				// Send nil to channel to indicate error
+				log.Error("could not get delegations", "err", err, "relay", relay.Config().Endpoint)
+				delegationsCh <- nil
+			}
+
+			delegationsCh <- &delegations
+		}(relay, i)
+	}
+
+	err := errors.New("could not get delegations from any relay")
+
+	aggregated := types.SignedDelegations{}
+	for i := 0; i < len(r.relays); i++ {
+		d := <-delegationsCh
```

Comment:
Is there a timeout for the `GetDelegationsForSlot`? Otherwise this might hang waiting for the some relays to respond.